README.md

```diff
@@ -1,58 +1,203 @@
+- The join server validates the join request (by validating app secret, captcha, or whatever is needed), and then makes a blockchain transaction to add the member to the Data Union.
 
-[Nvm](https://github.com/nvm-sh/nvm#readme) is used to install Node and npm. Node version is specified in [`.nvmrc`](.nvmrc).
+Data Union builder teams can easily [extend](#extending) the validation logic and run their own join server. Implementing any kind of join request validation logic is possible.
 
-### Common Commands
+As an alternative to running your own customized join server, the Data Union DAO hosts a **default join server**, which also extends this base package and implements a validation logic based on application secrets stored in a database.
+
+This package can also be [run as-is](#running-as-is), in which case the join server performs only signature validation and therefore allows anyone to join a Data Union.
+
+## Usage
+
+The most typical use case for this package is to extend the functionality of the join server by adding custom validation logic and/or additional HTTP endpoints.
+
+- Start a new node.js project for your custom join server
+- Install this base package as a dependency:
+
+```
+npm install --save @dataunions/join-server
+```
+
+- Then you can use the server in your application:
+
+```
+const { JoinServer } = require('@dataunions/JoinServer')
+
+const srv = new JoinServer({
+    // Always pass in the private key for the wallet you set as the joinPartAgent
+    privateKey: '...',
+
+    // Additional options, see below
+    ...
+})
+srv.start()
+```
+
+### Options
+
+See below for the various constructor options and their default values. At a minimum, you should pass in at least the `privateKey`.
+
+```
+new JoinServer({
+    // Hex-encoded private key for your joinPartAgent address
+    privateKey: '...', 
+
+    // HTTP port the server listens on
+    port: 5555,
+
+    // Logger (pino) level: one of 'fatal', 'error', 'warn', 'info', 'debug', 'trace' or 'silent'.
+    logLevel: 'info',
+
+    // Used to validate custom fields in join requests. The default function does nothing.
+    customJoinRequestValidator: async (joinRequest) => {},
+
+    // Used to add custom routes to the HTTP server. The default function does nothing.
+    customRoutes: (expressApp) => {},
+
+    // By default public RPCs are used for each chain, but you can pass this option to override
+    customRPCs: {
+        polygon: 'https://my-custom-polygon-rpc-address',
+        gnosis: 'https://my-custom-gnosis-rpc-address',
+    }
+})
+```
+
+## Running as-is
+
+You can also run the "base" join server without any customizations. This may be useful for development and testing. Note that the base join server only does the signature validation, meaning that anyone (including bots etc.) can join your data unions.
 
-Install dependencies from package.json to get started
 ```
-make npm-ci
+npm install -g @dataunions/join-server
```

Comment:
I guess `-g` is common in instructions, but I still find it better taste to not do global installs:
```
npm install @dataunions/join-server
npx join-server -k <private key>
```